src/signer_node/mod.rs

```diff
@@ -306,36 +325,49 @@ impl<T: TapyrusApi, C: ConnectionManager> SignerNode<T, C> {
         }
     }
 
-    pub fn process_message(&mut self, message: Message) -> NodeState {
-        match message.message_type {
-            MessageType::Candidateblock(block) => {
-                process_candidateblock(&message.sender_id, &block, self)
+    pub fn process_key_generation_message(
+        &mut self,
+        sender_id: SignerID,
+        message: KeyGenerationMessageType,
+    ) {
+        match message {
+            KeyGenerationMessageType::Nodevss(vss, secret_share) => {
+                process_nodevss(&sender_id, vss, secret_share, self);
             }
-            MessageType::Completedblock(block) => {
-                process_completedblock(&message.sender_id, &block, self)
+        }
+    }
+
+    pub fn process_round_message(
```

Comment:
Could you show me the reason?

Reply:
Because  sender_id is used as reference in this function.
By the way, i have written wrong signature of the function.
Fixed one is :

```
pub fn process_round_message(
        &mut self,
        sender_id: &SignerID,
        message: BlockGenerationRoundMessageType,
)
```